src/audio/AudioStream.ts

```diff
@@ -192,10 +197,26 @@ class AudioStream {
       const hlsConfig = { ...HlsConfig, fLoader: creatorFLoader }
       this.hls = new Hls(hlsConfig)
 
+      // On load of HLS, reset error rate limiter
+      this.errorRateLimiter = new Set()
+
       this.hls.on(Hls.Events.ERROR, (event, data) => {
+        // Only emit on fatal because HLS is very noisy (e.g. pauses trigger errors)
         if (data.fatal) {
-          // Only emit on fatal because HLS is very noisy (e.g. pauses trigger errors)
-          this.onError(AudioError.HLS, data)
+          // Buffer stall errors occur but are rarely fatal even if they claim to be.
+          // This occurs when you play a track, pause it, and then wait a few seconds.
+          // It appears as if we see this despite being able to play through the track.
+          if (data.details === 'bufferStalledError') {
+            return
+          }
+
+          // Only emit an error if we haven't errored on an error with the same "details"
+          if (!this.errorRateLimiter.has(data.details)) {
```

Comment:
i'm assuming there's no unique info per error in the details obj that we can safely de-dupe?

Reply:
yeah it's just a string. i haven't seen any w/ unique info, but if we find some, it's probably good to know via error reporting anyway